src/domain/usecases/actions/Action1UseCase.ts

```diff
@@ -8,11 +12,79 @@ export class Action1UseCase implements UseCase {
     constructor(private martRepository: XMartRepository, private instanceRepository: InstanceRepository) {}
 
     public execute(): FutureData<void> {
-        // TODO: Implement use-case logic
-        // Use mart repository to get the data from xMART
-        // Use instance repository to save the data to DHIS2
-        console.debug("Action 1", this.martRepository, this.instanceRepository);
+        return this.martRepository
+            .listAll("FACT_MOLECULAR_TEST")
+            .map(options => {
+                const events: ProgramEvent[] = _.compact(
+                    options.map(item => {
+                        const event = item["TEST_ID"] ? item["TEST_ID"] : getUid(String(item["_RecordID"]));
```

Comment:
Nullish coalescing operator:

```suggestion
                        const event = item["TEST_ID"] ?? getUid(String(item["_RecordID"]));
```